python/pyspark/sql/tests/test_functions.py

```diff
@@ -278,6 +279,22 @@ def test_sort_with_nulls_order(self):
             df.select(df.name).orderBy(functions.desc_nulls_last('name')).collect(),
             [Row(name=u'Tom'), Row(name=u'Alice'), Row(name=None)])
 
+    def test_input_file_name_reset_for_rdd(self):
+        from pyspark.sql.functions import udf, input_file_name
+        rdd = self.sc.textFile('python/test_support/hello/hello.txt').map(lambda x: {'data': x})
+        df = self.spark.createDataFrame(rdd, StructType([StructField('data', StringType(), True)]))
+        df.select(input_file_name().alias('file')).collect()
+
+        non_file_df = self.spark.range(0, 100, 1, 100).select(input_file_name().alias('file'))
+
+        results = non_file_df.collect()
+        self.assertTrue(len(results) == 100)
+
+        # [SC-12160]: if everything was properly reset after the last job, this should return
```

Comment:
What's SC-12160?

Reply:
+1 for @HyukjinKwon 's comment. Is this the internal issue tracker ID?
Could you update the PR, @jose-torres ?